src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -40,8 +42,9 @@ private void InitializeCancellationToken(TimeSpan timeout)
             }
             else if (timeout > TimeSpan.Zero)
             {
-                _cts = new CancellationTokenSource(timeout);
+                _cts = new CancellationTokenSource();
```

Comment:
I don't like having this extra CancellationTokenSource allocation. It's only needed to support the CancelCancellationToken method which I can only find a single usage of in HttpChannelFactory. This functionality should be either placed behind an overload or embedded into HttpChannelFactory itself. Otherwise we have an extra unnecessary CTS and CT being created every time. If you want to measure the perf impact, you can do a temporary experiment by removing the creation of the extra CTS and CT, remove the call to _cts.Dispose() in the Dispose method and make the CancellationTokenSource() method throw. HttpChannelFactory only calls this method in the Abort/Faulting case so you should be able to run your perf tests unaffected by the temporary change. This will give us numbers to show how much of an impact (and how important to fix) this is.